### DISCUSSION TITLE
Duda sobre TPA2

### DISCUSSION THREAD

**Discussion by Franpanozzo** [dds-jv/foro#265] [TPA - 2021] [answered]
Buenas, teníamos una duda con el ultimo requerimiento: _Se debe permitir que una persona busque a su mascota perdida en la plataforma y que pueda contactarse con el rescatista en caso de encontrarla._ 
 Nosotros modelamos que las mascotas perdidas pueden tener una chapita la cual tiene un codigoQR. El usuario también tiene un codigoQR entonces cuando el rescatista encuentra una mascota con chapita se puede buscar que usuario se relaciona con la mascota en cuestión a través del QR. Pero en el caso de que no la tenga, **como puede este buscar a su mascota en el sistema ??** en la vida real entendemos que con las fotos ya seria necesario pero en el sistema necesitamos tener algún tipo de relacion como con el codigoQR.

Si nos pueden resolver la duda buenisimo, saludos!!

**Comment by gastonprieto** [comment, marked as answer]
> Pero en el caso de que no la tenga, como puede este buscar a su mascota en el sistema ?? en la vida real entendemos que con las fotos ya seria necesario pero en el sistema necesitamos tener algún tipo de relacion como con el codigoQR.

No te pide que busques automático.

> Se debe permitir que una persona busque a su mascota perdida en la plataforma

Esto es manual. Como en el ejercicio de QMP que decía que se tiene que elegir el guardarropa y era arbitrario, bueno, acá es lo mismo. La persona hace la búsqueda manual en la interfaz gráfica scrolleando hasta que encuentra su mascota y ahí va a disparar una acción para completar sus datos y que el rescatista lo contacte.

---
> Nosotros modelamos que las mascotas perdidas pueden tener una chapita la cual tiene un codigoQR. El usuario también tiene un codigoQR entonces cuando el rescatista encuentra una mascota con chapita se puede buscar que usuario se relaciona con la mascota en cuestión a través del QR

Acá esto me suena medio redundante, si tenes tu mascota con QR (igual no se cuanto les conviene modelar esto, ya que es algo físico, y que sirve para llegar a un link) entonces deberías poder llegar al dueño, me parece media redundante la doble relación.

Igual esto chequeenlo con su ayudante.

**Comment by Franpanozzo** [comment]
Entonces, como en el ejercicio de QMP, creo que te referís también a cuando el usuario podía aceptar solicitudes arbitrariamente y habíamos dicho en clase que suponemos que el usuario ya sabe la supuesta propuesta, que esa parte de buscar la propuesta que quiere aceptar es parte de una abstracción anterior, en este caso la UI, entonces podemos directamente inyectarlo en el método, es correcto lo que digo??

Por otra parte lo de la doble relacion podria ser, lo que nos habia hecho ruido a nosotros es que el rescatista al rellenar el supuesto formulario ( que seria en nuestro dominio hacer new de la mascotaPerdida con sus dependencias)  como que no tenia sentido que rellene cargando el dueño dentro de la Chapita, como que ya lo conoceria el rescatista de alguna forma, nos parecio que tenia mas sentido que cargue el codigoQR por eso decidimos poner un codigoQR en la chapita en vez de directamente el dueño.

**Reply by gastonprieto** [reply]
> Entonces, como en el ejercicio de QMP, creo que te referís también a cuando el usuario podía aceptar solicitudes arbitrariamente y habíamos dicho en clase que suponemos que el usuario ya sabe la supuesta propuesta, que esa parte de buscar la propuesta que quiere aceptar es parte de una abstracción anterior, en este caso la UI, entonces podemos directamente inyectarlo en el método, es correcto lo que digo??

Si, pueden asumir que ya lo selecciono. Igual confirmenlo y evaluen con su ayudante esta parte de la solución

> Por otra parte lo de la doble relacion podria ser, lo que nos habia hecho ruido a nosotros es que el rescatista al rellenar el supuesto formulario ( que seria en nuestro dominio hacer new de la mascotaPerdida con sus dependencias) como que no tenia sentido que rellene cargando el dueño dentro de la Chapita, como que ya lo conoceria el rescatista de alguna forma, nos parecio que tenia mas sentido que cargue el codigoQR por eso decidimos poner un codigoQR en la chapita en vez de directamente el dueño.

Esto mirenlo con su ayudante, porque deberia estar mas al tanto de su solución para darles mejor feedback.

**Reply by Franpanozzo** [reply]
Perfecto gaston muchas gracias!
